docs/source/developers/python.rst

```diff
@@ -518,13 +518,11 @@ We set a number of environment variables:
 
 - the path of the installation directory of the Arrow C++ libraries as
   ``ARROW_HOME``
```

Comment:
I would add something like "(when using a conda environment, Arrow C++ is installed in the env directories)"

(to explain why we are setting it to `%CONDA_PREFIX` below)